pkg/operators/cluster-autoscaler-operator.go

```
@@ -110,3 +117,322 @@ var _ = Describe("Cluster autoscaler cluster operator status should", framework.
+		return fmt.Errorf("TLS handshake failed with cipher suite %s: %w (stderr: %s)", cipherSuite, err, stderr.String())
+	}
+
+	return nil
+}
+
+// expectHandshakeRejected returns a function suitable for Eventually that
+// succeeds only when the TLS handshake with the given cipher suite fails.
+func expectHandshakeRejected(probe *tlsProbe, ctx context.Context, cipherSuite string) func() error {
+	return func() error {
+		if err := probe.handshake(ctx, cipherSuite); err != nil {
+			return nil // Connection failed as expected
```

golangci-lint (2.12.2)

[error] error is not nil (line 314) but it returns nil

(nilerr)